### ISSUE TITLE
Can't donate from a different metamask account other than the one use originally for donatiosn

### ISSUE THREAD

**Issue by laurenluz** [Giveth/giveth-next#266]
I was signed in with my account "Lauren Luz" on Giveth next, the wanted to try testing as an anonymous donor from a different account. I signed out, cleared cache, changed the account in my metamask wallet then tried to donate anonymously, but in metamask it would try to make the donation come from my other addresss (Associated w Lauren Luz).

there was seemingly no way to change this.

the donation page:
![image.png](https://images.zenhubusercontent.com/5fdfac16a58523a866ddb68d/857bcc97-2ecb-40d9-9b8a-e369c966fa9d)

my wallet:
![image.png](https://images.zenhubusercontent.com/5fdfac16a58523a866ddb68d/2a256334-599d-4fa2-befb-43afef998569)

**Comment by RamRamez** [comment]
@laurenluz Your wallet is not connected. It seems you haven't allowed the app to connect to your current account or you've just created a new one, so the app tries to connect to the previous account.

**Comment by laurenluz** [comment]
yeah, I wasn't signed in on purpose and was trying to donate from a different wallet (other than the one I have created an accoount with). But even with a cleared cached, it wouldn't let me donate with the other wallet.

**Comment by RamRamez** [comment]
@laurenluz No, I mean you haven't allowed the app to connect to your MetaMask account

![image](https://user-images.githubusercontent.com/73441857/135770103-b83c2c02-e635-4c11-97dd-b7b43cf654fa.png)

**Comment by laurenluz** [comment]
Oh I see, well now I connected it to the app.... but it was very confusing - even for me, as a regular Giveth user...

Is it easy to add something to the DApp... like something that says "switch account" on the donation page that then gives the user another opportunity to sign in w the new account? Because I didn't realize I could (or had to) do that in metamask

**Comment by RamRamez** [comment]
@laurenluz User can change his/her wallet now, but yeah I agree, I don't like the App user experience either. It's confusing even for Giveth members. So decided to completely change user sign process and sync it with user wallet.

![image](https://user-images.githubusercontent.com/73441857/135770693-1c259ea7-244c-4dea-ab03-e39682912a12.png)

**Comment by laurenluz** [comment]
@RamRamez I like the idea of syncing sign-in w wallet!!

btw thought... I tried to use the "click here to use another wallet" thing and selected metamask... but it just *opened* metamask, showing my non-connected wallet, and didn't give me the prompt to sign to connect it

**Comment by RamRamez** [comment]
@laurenluz Please explain in more details, what did you want to do? and what did you expect to see?

Clicking "click here to use another wallet" is for changing between MetaMask and Torus wallet. If you wanna change your account in MetaMask, you should use MetaMask itself. Changing accounts doesn't need signing.

**Comment by laurenluz** [comment]
But this was what I was saying was my bug in the first place. I wanted to donate with a metamask account that was different from the one I usually used to sign in. Let's say A is my normal account and B is the one I wanted to use.

I had account B open in metamask, and was testing donating as an anonymous, not-signed-in donor.
When I went to donate, it wanted to send the funds from account A even though I had account B open. I tried to click "switch wallet"... and then selected metamask, and nothing changed. 

It would be nice if, in the case of a user having a "not connected" account open in metamask and trying to donate... that they had an oppotunity/prompt to connect w metamask. Does that make more sense?

**Comment by RamRamez** [comment]
@laurenluz If you do not let the App to connect to your MetaMask account, the App can't do anything. I mean until your account state is "not connected", the MetaMask doesn't give us that account info (like balance). In this case, MetaMask gives the App, last connected account info instead. In your case, it would be account A.

It might be a little complicated and I don't know if I could explain it well to you, but if you think there is still something wrong with the donate page or I didn't get your point, let's have a talk on Discord.

**Comment by laurenluz** [comment]
yes but the real issue is that when I connected initally... it only prompted me to connect my first account. Then later, when I was signed out... there was no prompt to connect the new, not connected, wallet. the user would need to know to manually connect their account from metamask, which I think is confusing.
